File: Matrix Diagonal Sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int sum=0; // for storing the ans (sum)
        int n=mat.size(); //for rows
        int m=mat[0].size();// for column
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(i==j || (i+j)==mat.size()-1)
                    sum+=mat[i][j]; 
            }
        }
        return sum;
    }
};

File: Matrix Diagonal Sum.java
class Solution {
public int diagonalSum(int[][] mat) {
int n=mat.length;
int s = 0; //starting index for each row element
int e = mat.length-1; //last elemetn for each row element
int sum=0;

    for(int[] arr: mat)
    {
        sum+=arr[s]+arr[e];
        s++;
        e--;
    }
    if(n*n%2==0){   //if matrix has no common element(n*n is even)
        return sum;
    }
    else{
        sum = sum - mat[n/2][n/2]; //if matrix has common element
        return sum;
    }
        
    
}
}
